library/Vanilla/Models/Model.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPLv2
+ */
+
+namespace Vanilla\Models;
+
+use Garden\Schema\Schema;
+use Garden\Schema\ValidationException;
+use Vanilla\InjectableInterface;
+
+/**
+ * Basic model class.
+ */
+class Model implements InjectableInterface {
+
+    /** @var \Gdn_Database */
+    private $database;
+
+    /** @var Schema */
+    private $readSchema;
+
+    /** @var string */
+    private $table;
+
+    /** @var Schema */
+    private $writeSchema;
+
+    /**
+     * Basic model constructor.
+     *
+     * @param string $table Database table associated with this resource.
+     */
+    public function __construct(string $table) {
+        $this->table = $table;
+    }
+
+    /**
+     * Get resource rows from a database table.
+     *
+     * @param array $where Conditions for the select query.
+     * @param array $options Options for the select query.
```

Comment:
It's nice to list our options in the docs when we have an unknown set of keys allowed.